src/wh_client_crypto.c

```diff
@@ -0,0 +1,697 @@
+        uint16_t req_len = WH_PACKET_STUB_SIZE + sizeof(*req) + in_len;
+        uint32_t options = 0;
+
+        if (req_len <= WOLFHSM_CFG_COMM_DATA_LEN) {
+            if (evict != 0) {
+                options |= WH_PACKET_PK_ECSIGN_OPTIONS_EVICT;
+            }
+            req->type = type;
+            req->options = options;
+            req->keyId = key_id;
+            req->sz = in_len;
+            if( (in != NULL) && (in_len > 0)) {
+                XMEMCPY(req_in, in, in_len);
+            }
+
+            /* Send Request */
+            ret = wh_Client_SendRequest(ctx, group, action, req_len,
+                    (uint8_t*)packet);
+            if (ret == WH_ERROR_OK) {
+                /* Server will evict at this point. Reset evict */
+                evict = 0;
+
+                /* Response Message */
+                wh_Packet_pk_ecc_sign_res* res;
+                res = &packet->pkEccSignRes;
+                uint8_t* res_out = (uint8_t*)(res + 1);
+                uint16_t res_len;
+
+                /* Recv Response */
+                do {
+                    ret = wh_Client_RecvResponse(ctx, &group, &action, &res_len,
+                        (uint8_t*)packet);
+                } while (ret == WH_ERROR_NOTREADY);
+
+                if (ret == WH_ERROR_OK) {
+                    if (packet->rc == 0) {
+                        uint16_t out_len = res->sz;
+                        /* check inoutlen and read out */
+                        if (inout_len != NULL) {
+                            if (out_len > *inout_len) {
+                                /* Silently truncate the signature */
+                                out_len = *inout_len;
+                            }
+                            *inout_len = out_len;
+                            if (    (out != NULL) &&
+                                    (out_len > 0)) {
+                                memcpy(out, res_out, out_len);
+                            }
+                        }
+                    } else {
+                        ret = packet->rc;
+                    }
+                }
+            }
+        } else {
+            /* Request length is too long */
+            ret = WH_ERROR_BADARGS;
+        }
+    }
+    /* Evict the key manually on error */
+    if(evict != 0) {
+        (void)wh_Client_KeyEvict(ctx, key_id);
+    }
+#ifdef DEBUG_CRYPTOCB_VERBOSE
+    printf("[client] %s ret:%d\n", __func__, ret);
+#endif
+    return ret;
+}
+
+int wh_Client_EccDsaVerify(whClientContext* ctx, ecc_key* key,
+        const uint8_t* sig, word32 sig_len,
+        const uint8_t* hash, word32 hash_len,
+        int *out_res)
+{
+    int ret = 0;
+    whPacket* packet;
+
+    /* Transaction state */
+    whKeyId key_id;
+    int evict = 0;
+    int export_pub_key = 0;
+
+
+#ifdef DEBUG_CRYPTOCB_VERBOSE
+printf("[client] %s ctx:%p key:%p, in:%p in_len:%u, out:%p inout_len:%p\n",
+        __func__, ctx, key, in, in_len, out, inout_len);
+#endif
```

Comment:
Copy paste causing a build error - if you build with `DEBUG_CRYTOCB_VERBOSE` this will cause a compile error. `in`, `in_len`, `out`, `out_len` don't exist in this function. Build with this option to see if anything else breaks too

Reply:
will do.